lib/cell.rb

```diff
@@ -1,13 +1,14 @@
 class Cell
-  attr_reader :coordinate, :ship
+  attr_reader :coordinate, :ship, :fired_upon
 
-  def initialize(coordinate_parameter)
+  def initialize(coordinate_parameter, ship = nil)
     @coordinate = coordinate_parameter
-    @ship = nil
```

Comment:
So, the added code in line 4 works as a way to set a default value, but putting it here also allows a cell object to be initialized with a ship argument--which isn't something the interaction pattern calls for, so maybe we should avoid it. I'd recommend going back to the way it was before.